File: src/main/java/io/myosotisdev/utopianism/modules/Economy.java
package io.myosotisdev.utopianism.modules;

public class Economy
{
    private double money;

    public Economy()
    {
        this.money = 0;
    }

    public Economy(double money)
    {
        this.money = money;
    }

    public void giveMoney(double amount)
    {
        this.money += amount;
    }

    public void setMoney(double amount)
    {
        this.money = amount;
    }

    public boolean takeMoney(double amount)
    {
        if(this.money < amount)
        {
            return false;
        }
        this.money -= amount;

        return true;
    }

    public double getMoney()
    {
        return this.money;
    }

    public void deposit(double amount)
    {
        this.money += amount;
    }

    public boolean withdraw(double amount)
    {
        if(amount > this.money)
        {
            return false;
        }

        this.money -= amount;
        return true;
    }
}

File: src/main/java/io/myosotisdev/utopianism/modules/model/Economy.kt
package io.myosotisdev.utopianism.modules.model

import io.myosotisdev.utopianism.modules.player.PlayerEx
import io.myosotisdev.utopianism.modules.player.money
import net.minestom.server.entity.Player

class Economy
{
    var money: Double

    constructor()
    {
        money = 0.0
    }

    constructor(money: Double)
    {
        this.money = money
    }

    fun giveMoney(amount: Double)
    {
        money += amount
    }

    fun takeMoney(amount: Double): Boolean
    {
        if (money < amount)
        {
            return false
        }
        money -= amount
        return true
    }

    fun deposit(amount: Double)
    {
        money += amount
    }

    fun withdraw(amount: Double): Boolean
    {
        if (amount > money)
        {
            return false
        }
        money -= amount
        return true
    }
}